File: CardPuzzleSwift/CardPuzzleSwift/CardFlipSolutionFactory.swift
//
//  FlipCardSolutionFactory.swift
//  CardPuzzleSwift
//
//  Created by luke.bae on 2016. 9. 28..
//  Copyright © 2016년 Luke Bae. All rights reserved.
//

import Foundation


class CardFlipSolutionFactory {

    enum SolutionType {
        case PrimeNumberSolution
        case MultiplesSolution
        case BruteForceSolution
    }

    static func create(type:SolutionType) -> CardFlipSolution? {
        switch type {
        case .PrimeNumberSolution:
            return PrimeNumberFlip()
        case .MultiplesSolution:
            return MultiplesFlip()
        default:
            return nil
        }
    }
}

File: CardFlipPuzzle/CardFlipPuzzle/src/main/java/com/thewhalegames/puzzle/puzzle/CardFlipSolutionFactory.java
package com.thewhalegames.puzzle.puzzle;


import com.thewhalegames.puzzle.puzzle.solution.CardFlipSolution;
import com.thewhalegames.puzzle.puzzle.solution.MultiplesFlip;
import com.thewhalegames.puzzle.puzzle.solution.PrimeNumberFlip;
import com.thewhalegames.puzzle.puzzle.solution.SolutionType;

/**
 * Created by luke.bae on 2016. 10. 1..
 */
public class CardFlipSolutionFactory {

    static CardFlipSolution create(SolutionType type) {
        switch (type) {
            case MultiplesFlip:
                return new MultiplesFlip();
            case PrimeNumberFlip:
                return new PrimeNumberFlip();
            default:
                throw new RuntimeException("There is no such type: " + type);
        }
    }

}
